File: ios/ActionCableApp/Enums/DogName.swift
enum DogName: String {
    case chiyo = "chiyo"
    case eru = "eru"
    case otome = "otome"
    
    var capitalized: String {
        return self.rawValue.capitalized
    }
    
    init?(account: String) {
        switch account {
        case DogName.chiyo.rawValue:
            self = .chiyo
        case DogName.eru.rawValue:
            self = .eru
        case DogName.otome.rawValue:
            self = .otome
        default:
            return nil
        }
    }
    
    static var accounts: Array<String> {
        return [DogName.chiyo.rawValue, DogName.eru.rawValue, DogName.otome.rawValue]
    }
}

File: android/app/src/main/java/com/devnokiyo/actioncableapp/enums/DogName.kt
package com.devnokiyo.actioncableapp.enums


enum class DogName(val rawValue: String) {
    Chiyo("chiyo"),
    Eru("eru"),
    Otome("otome");

    fun capitalized(): String {
        return rawValue.capitalize()
    }

    companion object {
        var accounts: Array<String> = arrayOf()
            get() = arrayOf(DogName.Chiyo.rawValue, DogName.Eru.rawValue, DogName.Otome.rawValue)

        fun create(rawValue: String): DogName {
            return values().first { it.rawValue == rawValue }
        }
    }
}
